Bottom: f4b867a8d8edce730c38233b64ba965de5fcc92d
Top:    0a4930a4d309d6c541918925483a5751f70b043c
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-05 16:31:26 +0000

Adjust coordinate system GOODNES KNOWS WHETHER THE GEOMETRY IS CORRECT... IDA-STEP SEEMS TO COPE THOUGH


---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index f79ec16..6750e9c 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -24,6 +24,7 @@
 #define PERFECT_ROUND_CONTOURS
 
 #define REVERSED_PCB_CONTOURS 1 /* PCB Contours are reversed from the expected CCW for outer ordering - once the Y-coordinate flip is taken into account */
+#undef REVERSED_PCB_CONTOURS
 
 #ifdef REVERSED_PCB_CONTOURS
 #define COORD_TO_STEP_X(pcb, x) (COORD_TO_MM(                   (x)))
@@ -376,7 +377,7 @@ object3d_from_contours (const POLYAREA *contours,
 #ifdef REVERSED_PCB_CONTOURS
     face3d_set_normal (faces[npoints], 0., 0., -1.);
 #else
-    face3d_set_normal (faces[npoints], 0., 0., 1.);
+    face3d_set_normal (faces[npoints], 0., 0., 1.); /* PCB bottom is at positive Z in this scheme */
 #endif
     face3d_set_appearance (faces[npoints], top_bot_appearance);
     object3d_add_face (object, faces[npoints]);
@@ -385,7 +386,7 @@ object3d_from_contours (const POLYAREA *contours,
 #ifdef REVERSED_PCB_CONTOURS
     face3d_set_normal (faces[npoints + 1], 0., 0., 1.);
 #else
-    face3d_set_normal (faces[npoints + 1], 0., 0., -1.);
+    face3d_set_normal (faces[npoints + 1], 0., 0., -1.); /* PCB top is at negative Z in this scheme */
 #endif
     face3d_set_appearance (faces[npoints + 1], top_bot_appearance);
     object3d_add_face (object, faces[npoints + 1]);
diff --git a/src/hid/step/object3d_step.c b/src/hid/step/object3d_step.c
index aa2ee46..5b2cd40 100644
--- a/src/hid/step/object3d_step.c
+++ b/src/hid/step/object3d_step.c
@@ -25,7 +25,8 @@
 #include "object3d_step.h"
 
 
-#define REVERSED_PCB_CONTOURS 1 /* PCB Contours are reversed from the expected CCW for outer ordering - once the Y-coordinate flip is taken into account */
+//#define REVERSED_PCB_CONTOURS 1 /* PCB Contours are reversed from the expected CCW for outer ordering - once the Y-coordinate flip is taken into account */
+#undef REVERSED_PCB_CONTOURS
 
 #define EPSILON 1e-5 /* XXX: Unknown  what this needs to be */
